clickhouse/base/socket.cpp

```diff
@@ -390,9 +390,9 @@ std::unique_ptr<OutputStream> Socket::makeOutputStream() const {
 
 NonSecureSocketFactory::~NonSecureSocketFactory()  {}
 
-std::unique_ptr<SocketBase> NonSecureSocketFactory::connect(const ClientOptions &opts) {
-    const auto address = NetworkAddress(opts.host, std::to_string(opts.port));
+std::unique_ptr<SocketBase> NonSecureSocketFactory::connect(const ClientOptions &opts, const std::shared_ptr<EndpointsIteratorBase> endpoints_iterator) {
```

Comment:
What do you think about passing `NetworkAddress` object to this function instead of shared pointer to iterator ? Iterator seems redundant here

Reply:
Agree, fixed